src/App/Document.h

```diff
@@ -268,6 +268,21 @@ class AppExport Document: public App::PropertyContainer
                               bool isNew = true,
                               const char* viewType = nullptr,
                               bool isPartial = false);
+    //@{
+    /** Add a feature of T type with sName (ASCII) to this document and set it active.
+     * Unicode names are set through the Label property.
+     * @param pObjectName if nonNULL use that name otherwise generate a new unique name based on the
+     * \a sType
+     * @param isNew       if false don't call the \c DocumentObject::setupObject() callback (default
+     * is true)
+     * @param viewType    override object's view provider name
+     * @param isPartial   indicate if this object is meant to be partially loaded
+     */
+    template<typename T>
+    T* addObject(const char* pObjectName = nullptr,
+                 bool isNew = true,
+                 const char* viewType = nullptr,
+                 bool isPartial = false);

```

Comment:
Would be also good to replace `const char*` with `std::optional<const std::string>` on later PR as this contains quite a lot of `c_str()` on call-site.

Reply:
I agree. This is probably something we should try to do for a lot of places in the code base.